src/NServiceBus.Core/LogicalAddress.cs

```diff
@@ -10,28 +10,66 @@
     public struct LogicalAddress
     {
         /// <summary>
-        /// Creates new qualified logical address for the provided endpoint instance name.
+        /// Creates new root logical address for the provided endpoint instance name.
         /// </summary>
         /// <param name="endpointInstance">The name of the instance.</param>
-        /// <param name="qualifier">The qualifier of this address.</param>
-        public LogicalAddress(EndpointInstance endpointInstance, [NotNull] string qualifier)
+        public LogicalAddress(EndpointInstance endpointInstance)
+            : this(endpointInstance, null)
         {
-            if (qualifier == null)
-            {
-                throw new ArgumentNullException(nameof(qualifier));
-            }
-            EndpointInstance = endpointInstance;
-            Qualifier = qualifier;
         }
 
         /// <summary>
         /// Creates new root logical address for the provided endpoint instance name.
         /// </summary>
-        /// <param name="endpointInstance">The name of the instance.</param>
-        public LogicalAddress(EndpointInstance endpointInstance)
+        /// <param name="queueName">Name of the queue.</param>
+        /// <param name="discriminator">Discriminator for the queue.</param>
+        /// <param name="properties">Properties.</param>
+        public LogicalAddress(string queueName, string discriminator, IReadOnlyDictionary<string, string> properties)
+            : this(new EndpointInstance(queueName, discriminator, properties))
+        {
+        }
+
+        LogicalAddress(EndpointInstance endpointInstance, string qualifier)
         {
             EndpointInstance = endpointInstance;
-            Qualifier = null;
+            Qualifier = qualifier;
+        }
+
+        /// <summary>
+        /// Creates an address scoped inside this one.
+        /// </summary>
+        /// <param name="qualifier">The qualifier for the new address.</param>
+        public LogicalAddress CreateSubscope(string qualifier)
```

Comment:
Like the Set methods before, I think that this method and `CreateIndividualized` would be better if they were static methods where you passed in an existing instance. That makes it more obvious that you are creating a new instance that you need to store.